core/src/main/java/nl/nn/adapterframework/scheduler/job/CleanupDatabaseJob.java

```diff
@@ -162,11 +165,15 @@ public void execute() {
 
 				boolean deletedAllRecords = false;
 				while(!deletedAllRecords) {
-					Message result = qs.sendMessageOrThrow(Message.nullMessage(), null);
-					String resultString = result.asString();
-					log.info("deleted [" + resultString + "] rows");
-					int numberOfRowsAffected = Integer.valueOf(resultString);
-					if(maxRows<=0 || numberOfRowsAffected<maxRows) {
+					int numberOfRowsAffected = 0;
+					try (Message result = qs.sendMessageOrThrow(Message.nullMessage(), null)) {
+						String resultString = result.asString();
+						log.info("deleted [{}] rows", resultString);
+						if (NumberUtils.isDigits(resultString)) {
+							numberOfRowsAffected = Integer.parseInt(resultString);
+						}
```

Comment:
This changes the semantics of the code and could have unintended consequences.

Reply:
That's how I solved a null check on `resultString`, because it can be null.

Reply:
Yes, but if I look at the flow of the code, the flow has changed and this could have unintended consequences (potentially dangerous in this case).
I think that if there's NULL or non-numeric input, we should abort instead of continue like happens now.

Reply:
Ah ok, got it. Will make it like that. Thanks